### ISSUE TITLE
Captive portal behaviour

### ISSUE THREAD

**Issue by chris-box** [tfpauly/draft-pauly-adaptive-dns-privacy#67]
When reading the drafts it wasn't clear to me how the Hostname Resolution Algorithm is intended to work for a client sitting behind a captive portal.

One option is that Adaptive DNS is not intended to come into play until after the captive portal process has been completed.

Another option is that it should always be used. When "captive" I would expect this to result in failure of step 1 (Exclusive Direct Resolver). Would step 2 (Direct Resolver) sometimes work, for example if the local router published a domain that was also the one the client was asking for? If it didn't work, we carry on through and know that on reaching step 5 then it should work, assuming the captive portal detection domain should never be thought of as privacy-sensitive.

**Comment by tfpauly** [comment]
We can certainly clarify this in the resolution algorithm section. For our client (iOS/macOS), we don't do any user-specific traffic until we check if we are captive or not. The system traffic to interact with the captive portal MUST use the local resolver.

We do have some text in the section "Walled-Garden and Captive Network Deployments":

```
However, if such a network does not support explicitly advertising local names, clients that try to establish connections to DoH servers will experience connection failures. In these cases, system traffic that is used for connecting to captive portals SHOULD use local resolvers. In addition, clients MAY choose to fall back to using direct resolution without any encryption if they determine that all connectivity is blocked otherwise. Note that this comes with a risk of a network blocking connections in order to induce this fallback behavior, so clients might want to inform users about this possible attack where appropriate, or prefer to not fall back if there is a concern about leaking user data.
```

**Comment by chris-box** [comment]
Yes, the walled garden section does help, but as it's later and more narrative, it still leaves some uncertainty in my mind. Ideally it would be clearly described in section 3.3.

**Comment by tfpauly** [comment]
Addressed in 138dc9b
